apptools/logger/agent/attachments.py

```diff
@@ -58,7 +58,7 @@ def _attach_directory(self, dir):
         relpath = os.path.basename(dir)
 
         import zipfile
-        from cStringIO import StringIO
+        from six.moves import cStringIO as StringIO
```

Comment:
Would the simpler `from six import StringIO` work here?

Reply:
Actually, looking at the place where this is used, do we need `BytesIO` rather than `StringIO`?

Reply:
Looks like we do. On Python 3, working in a directory that contains a `tox.ini` file, I see this:
```
>>> z = zipfile.ZipFile(StringIO(), "w")
>>> z.write("tox.ini")
Exception ignored in: <function ZipFile.__del__ at 0x1076fe7b8>
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/zipfile.py", line 1789, in __del__
    self.close()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/zipfile.py", line 1807, in close
    self._write_end_record()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/zipfile.py", line 1911, in _write_end_record
    self.fp.write(endrec)
TypeError: string argument expected, got 'bytes'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/zipfile.py", line 1743, in write
    with open(filename, "rb") as src, self.open(zinfo, 'w') as dest:
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/zipfile.py", line 1470, in open
    return self._open_to_write(zinfo, force_zip64=force_zip64)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/zipfile.py", line 1583, in _open_to_write
    self.fp.write(zinfo.FileHeader(zip64))
TypeError: string argument expected, got 'bytes'
```